persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/member/MemberDAO.kt

```diff
@@ -176,11 +176,18 @@ internal class MemberDAOImpl internal constructor(
     override suspend fun updateFullMemberList(memberList: List<MemberEntity>, conversationID: QualifiedIDEntity) =
         withContext(coroutineContext) {
             memberQueries.transaction {
-                memberQueries.deleteMembersFromConversation(conversationID)
-                for (member: MemberEntity in memberList) {
+                val membersToDelete = memberQueries.selectAllMembersByConversation(conversationID)
+                    .executeAsList()
+                    .filter { member -> memberList.none { it.user == member.user } }
+
+                memberList.forEach { member ->
                     userQueries.insertOrIgnoreUserId(member.user)
                     memberQueries.insertMember(member.user, conversationID, member.role)
                 }
+
+                membersToDelete.forEach { member ->
+                    memberQueries.deleteMember(conversationID, member.user)
+                }
```

Comment:
Can we cover it with test ?

Reply:
your message helped me to understand that my implementation doesn't update existed users (if updating is needed). 
So i've added a new DB-query and update unit-test to cover that case too. 
Thanks :)